cmd/extensions/oidc-policy/internal/controller/oidcpolicy_reconciler.go

```diff
@@ -138,37 +135,39 @@ func (r *OIDCPolicyReconciler) Reconcile(ctx context.Context, request reconcile.
 	return reconcile.Result{}, nil
 }
 
-func (r *OIDCPolicyReconciler) reconcileSpec(ctx context.Context, pol *kuadrantv1alpha1.OIDCPolicy, igw *ingressGatewayInfo) error {
+func (r *OIDCPolicyReconciler) reconcileSpec(ctx context.Context, pol *kuadrantv1alpha1.OIDCPolicy, igw *ingressGatewayInfo) ([]*kuadrantv1.AuthPolicy, error) {
 	// Reconcile AuthPolicy for the oidc policy http route
-	if err := r.reconcileMainAuthPolicy(ctx, pol, igw); err != nil {
+	mainAuthPol, err := r.reconcileMainAuthPolicy(ctx, pol, igw)
+	if err != nil {
 		r.Logger.Error(err, "Failed to reconcile main AuthPolicy")
-		return err
+		return nil, err
 	}
 
 	// Reconcile HTTPRoute for the callback for exchanging code/token
-	if err := r.reconcileCallbackHTTPRoute(ctx, pol, igw); err != nil {
+	if err = r.reconcileCallbackHTTPRoute(ctx, pol, igw); err != nil {
 		r.Logger.Error(err, "Failed to reconcile callback HTTPRoute")
-		return err
+		return nil, err
 	}
 	// Reconcile AuthPolicy for the Token exchange flow with metadata http call
-	if err := r.reconcileCallbackAuthPolicy(ctx, pol, igw); err != nil {
+	callbackPol, err := r.reconcileCallbackAuthPolicy(ctx, pol, igw)
+	if err != nil {
 		r.Logger.Error(err, "Failed to reconcile callback AuthPolicy")
-		return err
+		return nil, err
 	}
 
-	return nil
+	return []*kuadrantv1.AuthPolicy{mainAuthPol, callbackPol}, nil
```

Comment:
I don't mind this as such but the API of this function does seem weird, returning a list of policies.. would an alternative be to be calculating the status step-by-step as you create the resources and return that instead?

Reply:
Yup, the alternative would be calculating and updating one by one, which would mean a couple of added set of queries to the cluster. Would you reckon that to be more suitable ?

Reply:
oh, gotcha, you mean just the validation/calculation not updating the resource

Reply:
Nice, the direction you took looks really good